packages/core/src/test/amazonqFeatureDev/controllers/chat/controller.test.ts

```diff
@@ -519,4 +557,36 @@ describe('Controller', () => {
             assertTelemetry('ui_click', { elementId: 'amazonq_stopCodeGeneration' })
         })
     })
+
+    describe('closeSession', async () => {
+        async function closeSessionClicked() {
+            const getSessionStub = sinon.stub(controllerSetup.sessionStorage, 'getSession').resolves(session)
+
+            controllerSetup.emitters.followUpClicked.fire({
+                tabID,
+                followUp: {
+                    type: FollowUpTypes.CloseSession,
+                },
+            })
+
+            // Wait until the controller has time to process the event
+            await waitUntil(() => {
+                return Promise.resolve(getSessionStub.callCount > 0)
+            }, {})
+        }
+
+        it('end chat telemetry is sent', async () => {
+            await closeSessionClicked()
+
+            assertTelemetry('amazonq_endChat', { amazonqConversationId: conversationID, result: 'Succeeded' })
+        })
+
+        it('calls disableFileList', async () => {
+            const disableFileListSpy = sinon.stub(session, 'disableFileList').resolves()
+
+            await closeSessionClicked()
+
+            assert.ok(disableFileListSpy.calledOnce)
```

Comment:
this is a brittle and not very useful test. is it possible to test the actual state/results instead of asserting a spy